extensions/applicationinsights-dependencies-js/Tests/Unit/src/ajax.tests.ts

```diff
@@ -1909,6 +1909,55 @@ export class AjaxTests extends AITestClass {
             }
         });
 
+        this.testCase({
+            name: "Fetch: fetch keeps custom headers when correlation headers are discarded by dependencyListeners",
+            test: () => {
+                hookFetch((resolve) => {
+                    AITestClass.orgSetTimeout(function() {
+                        resolve();
+                    }, 0);
+                });
+
+                try {
+                    this._ajax = new AjaxMonitor();
+                    this._ajax.addDependencyListener((details: IDependencyListenerDetails) => {
+                        return false;
+                    });
+                    let appInsightsCore = new AppInsightsCore();
+                    let coreConfig = {
+                        instrumentationKey: "",
+                        disableFetchTracking: false,
+                        disableAjaxTracking: true
+                    };
+                    appInsightsCore.initialize(coreConfig, [this._ajax, new TestChannelPlugin()]);
+                    let fetchSpy = this.sandbox.spy(window, "fetch");
+
+                    // Setup
+                    let headers = new Headers();
+                    headers.append('My-Header', 'Header field');
+                    let init = {
+                        method: 'get',
+                        headers: headers
+                    };
+                    const url = 'https://httpbin.org/status/200';
+
+                    let headerSpy = this.sandbox.spy(this._ajax, "includeCorrelationHeaders");
+
+                    // Act
+                    Assert.ok(fetchSpy.notCalled);
+                    fetch(url, init);
+
+                    // Assert
+                    Assert.ok(fetchSpy.calledOnce);
+                    Assert.ok(headerSpy.calledOnce);
+                    Assert.deepEqual(init, headerSpy.returnValue || headerSpy.returnValues[0]);
+            } catch (e) {
+                    console && console.warn("Exception: " + e);
+                    Assert.ok(false, e);
```

Comment:
So a better pattern (if you really want to keep the exception handling -- which is ok because you have this assert) would be

```ts
Assert.ok(false, "Unexpected failure: "  dumpObj(e));
```

And just loose the console warn, as the dumpObj has special case handling for Error objects to provide all of the content (message and stack) it a slightly more friendly format